template/native_enum_type_traits.njk

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2017 The Bacardi Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef GEN_NATIVE_ENUM_TYPE_TRAITS_H_
+#define GEN_NATIVE_ENUM_TYPE_TRAITS_H_
+
+#include <napi.h>
+#include <string>
+#include <type_traits>
+#include "core/native_type_traits.h"
+#include "idl_enum_type.h"
+
+{% for enum in enums %}
+template <>
+struct NativeTypeTraits<IDL{{enum.name}}>
```

Comment:
IMHO, the enum type can be just unique enum type(because it's just string) and there is no reason to be NativeTypeTraits.
This generating code will generate C++ code per one enum type. It generates too much codes and can make build time increased.

We should check whether the type is enum or not in generator first.
And then, we make other logics fixed and just generate enumuration strings by generator.
For examples,
```
// This is not actual code, so, I ignores convention and grammar in my convenience.
{% if type == 'enum' %}
const char* validEnums = [
{% for each enum strings %}
    {%e enum_string %}
{% endfor %}
];
if (!IsValidEnum(validEnums, inputEnum)) return;
{% else %}
// Previsous codes
{% endif %}
```